### PR TITLE
feat: [IOCOM-875] SEND Timeline with new DS

### PR DESCRIPTION
**Pull request by Vangaorth**
⚠️ This PR depends on #5700  ⚠️

## Short description
This PR adds the SEND timeline to the SEND message details, with the new DS.

|Simple timeline|Long timeline|
|-|-|
|Simulator Screenshot - iPhone 15 - 2024-04-18 at 17 27 17|Simulator Screenshot - iPhone 15 - 2024-04-18 at 17 24 22|

## List of changes proposed in this pull request
- `TimelineListItem` component to handle both menu entry and bottom sheet 

## How to test
Using the io-dev-api-server, generate some SEND messages and check that the timeline works as expected.

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -2810,18 +2815,24 @@ features:
       infoSection:
         title: Informazioni sulla notifica
         iun: Codice IUN
+        iunAccessibility: Codice IUN, copia
+        iunSectionTitle: Notifica
       timeline:
+        goToSend: "Accedi a SEND"
+        menuTitle: Stato della notifica
+        newInfo: "Per vedere maggiori dettagli e le ricevute e le attestazioni opponibili a terzi, accedi alla versione web"
```
Comment (thisisjp) [inline locales/en/index.yml:2823]:
```suggestion
        newInfo: "Accedi alla versione web per vedere maggiori dettagli, le ricevute e le attestazioni opponibili a terzi."
```

File: locales/it/index.yml [hunk 2]
```diff
@@ -2810,18 +2815,24 @@ features:
       infoSection:
         title: Informazioni sulla notifica
         iun: Codice IUN
+        iunAccessibility: Codice IUN, copia
+        iunSectionTitle: Notifica
       timeline:
+        goToSend: "Accedi a SEND"
+        menuTitle: Stato della notifica
+        newInfo: "Per vedere maggiori dettagli e le ricevute e le attestazioni opponibili a terzi, accedi alla versione web"
```
Comment (thisisjp) [inline locales/it/index.yml:2823]:
```suggestion
        newInfo: "Accedi alla versione web per vedere maggiori dettagli, le ricevute e le attestazioni opponibili a terzi."
```

**Review by thisisjp — COMMENTED** [review]
small copy improvement

**Comment by forrest57** [comment]
image

as of right now, the sizing of the bottomSheet seems off, can you check if it happens on your end aswell?

**Comment by Vangaorth** [comment]
> as of right now, the sizing of the bottomSheet seems off, can you check if it happens on your end aswell?

Yes, you are right: on some devices and dependently upon timeline size, `useIOBottomSheetAutoresizableModal` does not compute its height properly. I switched to the fixed height bottom sheet, computing its sizing manually (with an upper limit in order not to cover the entire window's height)

File: ts/features/pn/components/TimelineListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+} from "../utils";
+import { trackPNTimelineExternal } from "../analytics";
+import { handleItemOnPress } from "../../../utils/url";
+import { useIOSelector } from "../../../store/hooks";
+import { pnFrontendUrlSelector } from "../../../store/reducers/backendStatus";
+import { Timeline, TimelineItemProps } from "./Timeline";
+
+const topBottomSheetMargin = 122;
+const timelineBottomMargin = 292;
+const timelineItemHeight = 70;
```
Comment (forrest57) [inline ts/features/pn/components/TimelineListItem.tsx:25]: Ui-wise it seems to work as intended, but may I suggest the usage of `useIOBottomSheetAutoresizableModal`?

Comment (Vangaorth) [reply ts/features/pn/components/TimelineListItem.tsx:25]: `useIOBottomSheetAutoresizableModal` does not work and it is buggy, as you shown before

**Review by forrest57 — APPROVED** [review]

**Pull request merged**